shared/actions/settings.js

```diff
@@ -113,23 +118,33 @@ function * refreshNotificationsSaga (): SagaGenerator<any, any> {
   }
 }
 
-function * notificationsSaga (): SagaGenerator<any, any> {
-  yield [
-    takeLatest(Constants.notificationsRefresh, refreshNotificationsSaga),
-    takeLatest(Constants.notificationsSave, saveNotificationsSaga),
-  ]
+function * deleteAccountForeverSaga (): SagaGenerator<any, any> {
+  try {
+    const username = yield select(state => state.config.username)
+    if (!username) {
+      throw new Error('Unable to delete account: not username set')
+    }
+    yield call(loginAccountDeleteRpcPromise)
+    yield put(setDeletedSelf(username))
+  } catch (err) {
+    // TODO hook into global error handler
+    console.error(err)
+  }
 }
 
 function * settingsSaga (): SagaGenerator<any, any> {
   yield [
-    call(notificationsSaga),
```

Comment:
my original intention was we'd have different sub sagas for the various sections so incase we wanted to split it up it'd be easier. so instead of this i was thinking just adding a `call(deleteSaga)` which has all the delete related sagas (and others) instead of this being one big list. But i'm fine with this also